crates/iota-genesis-builder/src/validator_info.rs

```diff
@@ -152,6 +152,10 @@ impl GenesisValidatorInfo {
             bail!("proof of possession is incorrect: {e}");
         }
 
+        if self.info.protocol_key() == self.info.network_key() {
+            bail!("protocol pubkey must be different from the network pubkey");
+        }
+
```

Comment:
if would probably move it before the rather expensive looking previous check

Reply:
It's not so expensive that one would notice a difference
I moved it below because that's also the order here
https://github.com/iotaledger/iota/blob/74a1c7f9b66d8afd97e8403758b28910fdb78aef/crates/iota-types/src/iota_system_state/iota_system_state_inner_v1.rs#L138-L150